lib/models/invitation.py

```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from requests import HTTPError
+
+from lib.api.checkr_api import CheckrAPI
+from lib.helpers.checkr_api_error import CheckerAPIFailedCreation
+
+
+class WorkLocation:
+    pass
+
+
+@dataclass
+class Invitation:
+    checkr_api: CheckrAPI
+    package: str
+    candidate_id: str
+    work_locations: list[dict[str, str]]
+
+    node: Optional[str] = None
+    tags: Optional[list[str]] = None
+    communication_types: Optional[list[str]] = None
+    uri: Optional[str] = None
+    object: Optional[str] = None
+    invitation_url: Optional[str] = None
+    status: Optional[str] = None
+    created_at: Optional[str] = None
+    expires_at: Optional[str] = None
+    deleted_at: Optional[str] = None
+    report_id: Optional[str] = None
+
+    id: Optional[int] = None  # From API Creation Response
+
+    def create(self, raise_on_failure: bool = True) -> bool:
+        if self.communication_types is None:
+            self.communication_types = ["email"]
```

Comment:
This looks weird. Should belong in the `__init__` function, no?

Reply:
It was actually suggested by pyCharm, Should I explicit call the __init__ method of invitation and put it there?

Reply:
Actually now I could do what I was trying to which is put a default value to communication_types. Like this: 

```
@dataclass
class Invitation:
    checkr_api: CheckrAPI
    candidate_id: str
    communication_types = ["email"]
    package: str
    work_locations: list[dict[str, str]]
    id: Optional[int] = None
    created_at: Optional[str] = None
    deleted_at: Optional[str] = None
    expires_at: Optional[str] = None
    invitation_url: Optional[str] = None
    node: Optional[str] = None
    object: Optional[str] = None
    report_id: Optional[str] = None
    status: Optional[str] = None
    tags: Optional[list[str]] = None
    uri: Optional[str] = None
```

Reply:
Yes, putting it in the dataclass definition is the best way of doing it.